docs/en/resources/tools/alloydb-nla.md

```diff
@@ -0,0 +1,50 @@
+---
+title: "alloydb-ai-nl"
+type: docs
+weight: 1
+description: > 
+  A "alloydb-ai-nl" tool leverages AlloyDB's AI functions to execute natural language questions against the database.
+---
+
+## About
+
+A `alloydb-ai-nl` tool leverages AlloyDB's AI functions to execute natural language questions against the database. It allows users to query database information using natural language instead of SQL. It's compatible with the following sources:
+- [alloydb-postgres](../sources/alloydb-pg.md)
+
+The tool uses AlloyDB's natural language processing capabilities to interpret questions and convert them into appropriate SQL queries, which are then executed against the database. TODO: link to AlloyDB's documentation.
+
+## Fields
+
+`nlConfig` is the name of the `nl_config` created in AlloyDB.
+
+`nlConfigParameters` are the list of the parameters and values for the AlloyDB [PSV (parameterized secure views)](!https://cloud.google.com/alloydb/docs/ai/use-psvs#sanitize_queries_with_parameterized_secure_views).
+
+When using this tool, all the PSV parameters should be from filled with values from an auth service or a bounded param. These parameters should not be visible to the LLM agent. Instead, the LLM will only see one argument when using this tool - `question`, with the description being "The natural language question to ask."
+
+## Example
+
+```yaml
+tools:
+  ask_questions:
+    kind: alloydb-ai-nl
+    source: my-alloydb-source
+    description: "Ask questions to check information about flights"
+    nlConfig: "cymbal_air_nl_config"
+    nlConfigParameters:
```

Comment:
the latest version of the documentation is referring to this as "context". I think we should update these fields to match.
```suggestion
    context: "cymbal_air_nl_config"
    contextParameters:
```